src/tests/functional/plugin/cpu/single_layer_tests/topk.cpp

```diff
@@ -3,104 +3,209 @@
 //
 
 #include "functional_test_utils/ov_tensor_utils.hpp"
-#include "common_test_utils/data_utils.hpp"
 #include "test_utils/cpu_test_utils.hpp"
-
 #include "shared_test_classes/base/ov_subgraph.hpp"
 #include "ngraph_functions/builders.hpp"
 
-using namespace ngraph;
 using namespace InferenceEngine;
 using namespace CPUTestUtils;
 using namespace ov::test;
 
 namespace CPULayerTestsDefinitions {
-using topKParams = std::tuple<
-    ngraph::element::Type, // data precision
-    InputShape, // data shape
-    std::int64_t, // axis
-    ngraph::opset4::TopK::Mode,
-    ngraph::opset4::TopK::SortType>;
-
-class TopKLayerCPUTest : public testing::WithParamInterface<topKParams>, public ov::test::SubgraphBaseTest, public CPUTestsBase {
+
```

Comment:
Does the test covers cases than for the same compiled model we have different number of top_k value (second TopK op input) each inference?

Reply:
Yes, random top_k value for each inference of dynamic shape has been covered since https://github.com/openvinotoolkit/openvino/pull/8474. And I've retained them. Further discussion is always welcomed. Thanks Dmitry!